reactor-netty-http/src/test/java/reactor/netty/http/server/HttpServerTests.java

```diff
@@ -1893,6 +1893,68 @@ public void channelRead(@NotNull ChannelHandlerContext ctx, @NotNull Object msg)
 				.isEqualTo("delay500delay1000");
 	}
 
+	@Test
+	public void testGracefulShutdownIssue3509() throws Exception {
+		CountDownLatch latch1 = new CountDownLatch(2);
+		CountDownLatch latch2 = new CountDownLatch(1);
+		CountDownLatch latch3 = new CountDownLatch(1);
+		LoopResources loop = LoopResources.create("testGracefulShutdownIssue3509");
+		this.disposableServer =
+				createServer().runOn(loop)
+				              .doOnConnection(c -> {
+				                  c.onDispose().subscribe(null, null, latch2::countDown);
+				                  latch1.countDown();
+				              })
+				              // Register a channel group, when invoking disposeNow()
+				              // the implementation will wait for the active requests to finish
+				              .channelGroup(new DefaultChannelGroup(new DefaultEventExecutor()))
+				              .route(r ->
+				                  r.get("/delay500", (req, res) -> res.sendString(Mono.just("delay500").delayElement(Duration.ofMillis(500))))
```

Comment:
Do you think using a `Mono.fromCallable` with a latch in the lambda or use `VirtualTimeScheduler` with `StepVerifier` to emulate the passage of time could speed up the runtime of this test? As I imagine, this test will execute always above 500ms due to this delay and if tests like this add up they can slow down the build significantly.

Reply:
fixed